src/usexp/lexer.mll

```diff
@@ -158,50 +282,52 @@ and escape_sequence mode = parse
   | newline
     { Lexing.new_line lexbuf;
       Newline }
-  | ['\\' '\'' '"' 'n' 't' 'b' 'r'] as c
+  | ['\\' '\'' '"' '%' '$' 'n' 't' 'b' 'r'] as c
     { let c =
         match c with
         | 'n' -> '\n'
         | 'r' -> '\r'
         | 'b' -> '\b'
         | 't' -> '\t'
+        | '$' when mode = New_syntax ->
```

Comment:
Not sure about what are the practices regarding polymorphic comparison here, but you can avoid using it here by matching on `(c, mode)` instead.